tests/trainer/test_optimizers.py

```diff
@@ -335,25 +354,155 @@ def test_configure_optimizers_with_frequency(tmpdir):
     model = EvalModelTemplate()
     model.configure_optimizers = model.configure_optimizers__multiple_optimizers_frequency
 
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        max_epochs=1
-    )
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
     result = trainer.fit(model)
     assert result
 
 
+def test_step_scheduling_for_multiple_optimizers_with_frequency(tmpdir):
```

Comment:
Have you thought about merging the 3 frequency tests into one with a `pytest.mark.parametrize` decorator? It should be possible

Reply:
I think it will increase complexity of the `parametrize` call and the test itself since the `configure_optimizers` method in each test is configured differently wrt the lr schedulers and optimizers. Also the args to lr schedulers will be different for `epoch` schedulers and `step` schedulers. I can give it a try if you think it'll be helpful.

Reply:
I suggested it because they are some long tests and if we could merge them they would look much tidier. At first glance I didn't see that many differences (you don't have to be too smart about how to merge them, you can use an `if` for what is different between them)

The choice of scheduler/optimizer in each test is not actually important, correct? just the index logic you implemented

Reply:
Cool, makes sense!

Reply:
Found a way to do it in https://github.com/PyTorchLightning/pytorch-lightning/pull/4868/commits/07bd1e01ae00b34de41dda29d3a99feed765ea02

Reply:
Awesome! Can you format it too?